lib/utils.js

```diff
@@ -0,0 +1,115 @@
+function getConfigurations(isCachingEnabled) {
+    // TODO: implement a proper caching mechanism
+    var configs = require("configs.json");
+    configs.isCachingEnabled = isCachingEnabled;
+    return configs;
+}
+
+function getLookUpTable(configs) {
+    // TODO: implement a proper caching mechanism
+    var log = new Log("utils-app-context");
+
+    function getLayoutsData(layoutsDir) {
+        var layoutsData = {};
+        var layoutsFiles = new File(layoutsDir).listFiles();
+        for (var i = 0; i < layoutsFiles.length; i++) {
+            var layoutFile = layoutsFiles[i];
+            if (layoutFile.isDirectory()) {
+                // this is not a layout, ignore
+                continue;
+            }
+
+            layoutsData[layoutFile.getName()] = {
+                path: layoutFile.getPath()
+            }
+        }
+        return layoutsData;
+    }
+
+    /**
+     * Returns unit's data.
+     * @param unitsDir {string} path to the units directory e.g. "/app/units"
+     * @return {{
+     *          units: Object<string, {name: string, path: string, definition: Object}>,
+     *          pushedUnits: Object<string, string[]>,
+     *          extendedUnits: Object.<string, string[]>}} unit data
+     */
+    function getUnitsData(unitsDir) {
+        /**
+         * @type {Object<string, {shortName: string, path: string, definition: Object}>}
+         */
+        var units = {};
+        /**
+         * @type {Object<string, string[]>}
+         */
+        var pushedUnits = {};
+        /**
+         * @type {Object<string, string[]>}
+         */
+        var extendedUnits = {};
+        var unitDirs = new File(unitsDir).listFiles();
+        for (var i = 0; i < unitDirs.length; i++) {
+            var unitDir = unitDirs[i];
+            if (!unitDir.isDirectory()) {
+                // this is not an unit, ignore
+                continue;
+            }
+
+            var unitFullName = unitDir.getName();
+            var unitShortName = unitFullName.split(".").pop();
+            var unitPath = unitsDir + "/" + unitFullName;
```

Comment:
Theoretically slash "/" should work on all major platforms. However UUF should be tested on all major platforms.